Source/Core/Core/IOS/USB/Emulated/LogitechMic.cpp

```diff
@@ -0,0 +1,737 @@
+// Copyright 2025 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "Core/IOS/USB/Emulated/LogitechMic.h"
+
+#include <algorithm>
+
+#include "Core/Config/MainSettings.h"
+#include "Core/HW/Memmap.h"
+#include "Core/System.h"
+
+namespace IOS::HLE::USB
+{
+enum class RequestCode : u8
+{
+  SetCur = 0x01,
+  GetCur = 0x81,
+  SetMin = 0x02,
+  GetMin = 0x82,
+  SetMax = 0x03,
+  GetMax = 0x83,
+  SetRes = 0x04,
+  GetRes = 0x84,
+  SetMem = 0x05,
+  GetMem = 0x85,
+  GetStat = 0xff
+};
+
+enum class FeatureUnitControlSelector : u8
+{
+  Mute = 0x01,
+  Volume = 0x02,
+  Bass = 0x03,
+  Midi = 0x04,
+  Treble = 0x05,
+  GraphicEqualizer = 0x06,
+  AutomaticGain = 0x07,
+  Delay = 0x08,
+  BassBoost = 0x09,
+  Loudness = 0x0a
+};
+
+enum class EndpointControlSelector : u8
+{
+  SamplingFreq = 0x01,
+  Pitch = 0x02
+};
+
+bool LogitechMicState::IsSampleOn() const
+{
+  return true;
+}
+
+bool LogitechMicState::IsMuted() const
+{
+  return mute;
+}
+
+u32 LogitechMicState::GetDefaultSamplingRate() const
+{
+  return DEFAULT_SAMPLING_RATE;
+}
+
+class MicrophoneLogitech final : public Microphone
+{
+public:
+  explicit MicrophoneLogitech(const LogitechMicState& sampler, u8 index)
+      : Microphone(sampler, GetWorkerName()), m_sampler(sampler), m_index(index)
+  {
+  }
+
+private:
+#ifdef HAVE_CUBEB
+  std::string GetWorkerName() const
+  {
+    return "Logitech USB Microphone Worker " + std::to_string(m_index);
```

Comment:
`m_index` isn't initialized yet when this function is invoked for initializing the base class.
You can just eliminate this `GetWorkerName` function and use `fmt::format("Logitech Mic {}", index)` in the `Microphone` initializer, which solves another issue, long thread names don't always display well. "Logitech Mic #" should be fine.